static/app/utils/replays/hooks/useExtractedPageHtml.tsx

```diff
@@ -0,0 +1,21 @@
+import {useQuery} from 'sentry/utils/queryClient';
+import extractPageHtml from 'sentry/utils/replays/extractPageHtml';
+import type ReplayReader from 'sentry/utils/replays/replayReader';
+
+interface Props {
+  offsetMsToStopAt: number[];
+  replay: ReplayReader | null;
+}
+
+export default function useExtractedPageHtml({replay, offsetMsToStopAt}: Props) {
+  return useQuery(
```

Comment:
why useQuery instead of useMemo

Reply:
useQuery will await the promise. It's a pattern i used before here: https://github.com/getsentry/sentry/blob/02f53d9b3a8251a0e108fd91104f529816e21a98/static/app/utils/replays/hooks/useExtractedDomNodes.tsx#L5-L17

Reply:
didn't even see the promise, that makes sense